src/modelframe.jl

```diff
@@ -141,14 +141,10 @@ function null_omit(df::DataTable)
     df[cc,:], cc
 end
 
-_droplevels!(x::Any) = x
-_droplevels!(x::Union{CategoricalArray, NullableCategoricalArray}) = droplevels!(x)
-
 function ModelFrame(trms::Terms, d::AbstractDataTable;
                     contrasts::Dict = Dict())
     df, msng = null_omit(DataTable(map(x -> d[x], trms.eterms)))
     names!(df, convert(Vector{Symbol}, map(string, trms.eterms)))
-    for c in eachcol(df) _droplevels!(c[2]) end
```

Comment:
@kleinschmidt Can you confirm this is OK? I've checked that unused levels continue to be skipped (since we call  `unique`), but I'd like to be sure I'm not missing something.